### ISSUE TITLE
[JAVA] JAXB models should not have to include JSON-related annotations

### ISSUE THREAD

**Issue by abstratt** [swagger-api/swagger-codegen#6054]
##### Description

I am testing the recently improved support for generating JAXB-based models (#5962 and #6005).
The generated code always includes JSON-related annotations, which are unnecessary for a XML-based API. 

##### Swagger-codegen version

2.2.3-SNAPSHOT

##### Configuration file

```json
{
  "sourceFolder" : "src/main/java",
  "withXml": true,
  "modelPackage" : "my.model.package",
  "additionalProperties" : { 
    "gson": false 
  }
}
```

##### Command line used for generation

```
    java -Dmodels -jar swagger-codegen-cli.jar \
        generate -i swagger.json  -l java -c config.json
```

##### Steps to reproduce

Run the generator using the provided configuration file and any Swagger spec. Observe the generated model classes end up with references to annotations and types from com.google.gson. It is fine if that is the default, but it should not be mandatory.

##### Suggest a Fix

I can change my configuration using the library option to choose a different JSON library. Maybe I should be able to specify "none"?

**Comment by wing328** [comment]
@abstratt are you saying if `withXML`is set to true, then `gson` in additional properties should automatically set to false as well?

cc @mobreza @packleader

**Comment by abstratt** [comment]
@wing238 I don't know about being automatically set to false, but at least it should not be automatically set to true as it is right now.  To be honest, the fact that I was not able to change the value for that property in my configuration file was really surprising. Defaults are fine when they make the user experience better, but it should not be at the expense of honoring user-provided configuration. If that approach was changed, then you would not have to do anything special for this use case. And would allow people that want both XML and JSON support to get it their way as well.

**Comment by mobreza** [comment]
@abstratt @wing328 The codegen primarily targeted `application/json` clients and support for XML was limited until #5962. That mostly addresses the configuration of the HTTP client and introduced some XML annotations.

Gson annotations are generated because the default library is the **okhttp-gson**. Switching to Jersey2 generates the model with `com.fasterxml.jackson` annotations.

```json
{
  "sourceFolder" : "src/main/java",
  "withXml": true,
  "modelPackage" : "my.model.package",
  "additionalProperties" : {
    "library": "jersey2" 
  }
}
```

Would that generate more appropriate model code for your use case? 

Do we need a `withJson` flag?

**Comment by abstratt** [comment]
@mobreza Yup, I am aware the XML support in the base Java generator is new, thank you guys for getting that done.

Now that XML is supported, swagger-codegen Java support handles JSON-only and JSON+XML-based REST APIs. Our case though is a XML-only REST API. I think you are pretty close to support that now, by somehow making JSON support optional.

An withJson flag, as you suggested, would do the job for us. If it defaulted to true, existing users would not be affected, and we would not have to add a spurious dependency on a JSON library to our XML-only model.

**Comment by mobreza** [comment]
@abstratt If you build the model separately from your main project, then the classes of Java annotations (gson/fasterxml) should not be required to build the main project with model as dependency. Would that work for you?

**Comment by abstratt** [comment]
@mobreza This issue is not a blocker, I do have a workaround (include the unnecessary dependency), and you are right, I can exclude if from downstream consumers if I am careful.

Still seems worth fixing, and hopefully a simple fix?  I understand that XML support is new, but hopefully it is here to stay, and worth evolving, instead of relegated to 2nd-class. My guess is that most APIs that support XML do not support JSON (support one is cheaper than supporting the two of them), so it is weird that a XML-only API has to have JSON-references in the source code.

**Comment by mobreza** [comment]
@abstratt It would be fairly simple to add (within Java) to any library specific generator  (e.g. jersey2) the `withJson` flag. 

Please see #5918 (esp https://github.com/swagger-api/swagger-codegen/issues/5918#issuecomment-317250708).
